rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -2055,13 +2075,35 @@ rmw_take_response(
 
   *taken = true;
 
-  for (z_owned_reply_t & reply : client_data->replies) {
```

Comment:
@clalancette looking back at this diff, I wonder if we should keep the original. If there is more than one server present, replies from other servers will still be stored in `client_data->replies`. We need to drop other replies to let the zenoh queryable that we're done with the reply?